### DISCUSSION TITLE
docker版本的aria2无法实现离线下载

### DISCUSSION THREAD

**Discussion by darkelfzdy** [AlistGo/alist#5907] [Q&A] [answered]
我用N1盒子刷的openwrt，alist是直接安转在openwrt上的，aria2用的这个docker版本（p3terx/aria2-pro）
alist的临时文件放在/mnt/sda4/alist_tmp，aria2的下载目录映射到/mnt/wky/down，在alist的其他-设置中aria2地址设置正常，能够连接识别。alist使用发送aria2下载也是正常的。
但如果使用alist离线下载就无法找到/mnt/sda4/alist_tmp，提示信息如下：
```
download http://192.168.31.97:5244/d/ali/media/动漫/托马斯和他的朋友们/第98集%20-%20油漆罐和皇后.flv to (/wky/down)
失败
[aria2]:

failed to transfer file: failed to get files: lstat /mnt/sda4/alist_tmp/aria2/b1e45be9-5631-4806-9abe-a2f755cad88e: no such file or directory
```
我怀疑是/mnt/sda4/alist_tmp这个文件夹的权限不对，但提权后还是不行，请问各位大佬知道是怎么回事吗？
补充：但如果使用openwrt上直接安装的aria2就一切正常。alist版本是3.30

**Comment by darkelfzdy** [comment]
我自己也在反复检查，发现是不是aria2在构建容器的时候设置了PUID=0 ，PGID=0，但如果不这么设置，就完全没法下载了

**Comment by darkelfzdy** [comment, marked as answer]
还是自己解决了。
仔细找了一下原因，网上有种说法无法transfer file，是因为aria2容器少了映射目录/opt/alist/data/temp/aria2，我按照这个网址的搞法：https://www.psay.cn/toss/216.html#:~:text=%E8%BF%99%E4%B8%AA%E5%8E%9F%E5%9B%A0%E4%B8%80%E8%88%AC%E6%98%AF%E7%94%B1,%E7%8E%B0%E4%B8%8A%E9%9D%A2%E7%9A%84%E9%94%99%E8%AF%AF%E4%BA%86%E3%80%82
在aria2中进行了映射 -v /mnt/sda4/alist_tmp/aria2:/opt/alist/data/temp/aria2，但还是不成功。
我突然想起，在我的openwrt上alist的临时目录不是/opt/alist/data/temp/aria2，而就是/mnt/sda4/alist_tmp/aria2，因此重新映射了一下，-v /mnt/sda4/alist_tmp/aria2:/mnt/sda4/alist_tmp/aria2，果然成功解决问题
这中间还是得到了多个人工智能的帮助，还是要靠自己呀

**Reply by xiayeroot** [reply]
其实就是aria2没有映射到alist的下载路径，aria2找不到alist指定的路径
